tests/mock_crust/churn.rs

```diff
@@ -550,13 +572,19 @@ fn aggressive_churn() {
         count_sections(&nodes)
     );
     while nodes.len() > target_network_size / 2 {
-        drop_random_nodes(&mut rng, &mut nodes, min_section_size);
+        let dropped_nodes = drop_random_nodes(&mut rng, &mut nodes, min_section_size);
 
         // A candidate could be blocked if it connected to a pre-merge minority section.
         // Or be rejected when the proxy node's RT is not large enough due to a lost tunnel.
         // In that case, a restart of candidate shall be carried out.
         if let Some(added_index) =
-            add_node_and_poll(&mut rng, &network, &mut nodes, min_section_size)
+            add_node_and_poll(
+                &mut rng,
+                &network,
+                &mut nodes,
+                min_section_size,
+                dropped_nodes,
```

Comment:
when we carry out check, we haven't poll yet, hence they are still in the RT.

if we don't exclude them, there will be un-necessary advance when only the dropped nodes being the RoutingConnection-proxy/joining